packages/grafana-ui/src/unstable.ts

```diff
@@ -0,0 +1 @@
+export * from './components/Text/TextElements';
```

Comment:
```suggestion
/**
 * THESE COMPONENTS MUST NOT BE USED EXTERNALLY OR IN PLUGINS.
 *
 * Unstable components are still in development and are subject to breaking changes
 * at any point, like feature flags but for components. They must only be used in
 * Grafana core where we can coordinate changes.
 *
 * Once mature, they will be moved to the main export, be available to plugins, and
 * be subject to the standard policies
 */

export * from './components/Text/Text';
export * from './components/Text/TextElements';

```

Reply:
@joshhunt thanks for this text. Good shout! 😄 Just one question, do we want to export the `Text component` too? I thought we agreed people should just use `<H1/>`, `<H2/>`, [...] 🤔

Reply:
Both!

```tsx
<H1>Hello <Text fontWeight="bold">Josh</H1>
```

Reply:
🤔 I think in that case it should be: 
```tsx
<H1>Hello <Span variant="h1" fontWeight="bold">Josh</Span></H1>
```
An 'element component', e.g. `<H1/>`, returns the `Text component` with a specific value for the `as` prop. We did not want to expose the `Text component` after creating `<H1/>`, `<H2/>`, [...]

Reply:
I don't think there's anything wrong about exposing `<Text />`, or calling it Text. I don't like the name `<Span />` because it's too.... HTML-y. In my example, I don't actually care about what element Text is, I just want text that is bold. I _do_ want an h1 that is h1-sized.

The point of `<H1 />` etc was to have 'sensible defaults' with an easy to read API.  They can totally co-exist with `<Text />` as a part of the public API.

Reply:
Hmm personally I don't like having 2 different ways to use the `Text` component, if we let people use both the `H1`, etc and `Text` we have basically gone for both options we were discussing and people can choose based on their preference. Unless there is a way for us to tell people NOT to use the `as` prop... Is a comment enough?

Reply:
@JoaoSilvaGrafana @eledobleefe 

Ahh, I understand now. I've just seen that we already have this `<Span />` element for styling text inline, so it doesn't make sense to expose Text. Not a fan of the name, but not worth holding up. Would much rather merge this in and iterate :) 

I put together this idea/proposal for how i thought about it https://github.com/grafana/grafana/pull/67480, but this is basically the same as what we have now (except effectively I renamed Span to Text)

Reply:
@joshhunt we've just created a `TextModifier` so now you can have "My name is **Josh**"  just with

```tsx
<H1>My name is <TextModifier weight="bold">Josh</TextModifier></H1>
```

Btw, good shout! I didn't realise about giving an option to modify the text inside a text component. Users should be able to hightlight a word by turning it to bold for example.

Reply:
@joshhunt what do you think of the changes we made? 
I took a look at https://github.com/grafana/grafana/pull/67480 and I like some things like the function to create the element components and also having all that is related to style in a file apart.
As we should have the `Text component` merged today, what do you think @joshhunt @JoaoSilvaGrafana about merging this PR and next week refactoring it taking into account Josh's ideas?

Reply:
I also like the changes you made Josh about having the styles separate and having that function create text components! I was about to apply that, but when I run it storybook doesnt show me a "Text" story, maybe because it doesn't deal very well with components that are created in that way?

So I think yes lets merge this as it is now (maybe adding that `TextModifier` component) and then we can think about refactoring it.

Reply:
sounds good. lets merge.

i also noticed I couldn't find the story, but hoped/assumed it wasnt a real issue (or caused by me lol).